File: app/src/main/java/hut/cwp/mvppresenter/MainPresenter.java
package hut.cwp.mvppresenter;

import android.os.Bundle;
import android.widget.Toast;

import mvp.MvpPresenter;

/**
 * Created by hutcwp on 2018/8/17 23:44
 * email: caiwenpeng@yy.com
 * YY: 909076244
 **/
public class MainPresenter extends MvpPresenter<IMain> {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    public void click(){
        getView().changeText();
    }
}

File: sample/src/main/java/hut/cwp/mvppresenter/MainPresenter.kt
package hut.cwp.mvppresenter

import android.os.Bundle
import android.util.Log
import hut.cwp.core.MvpPresenter

/**
 * Created by hutcwp on 2018/8/17 23:44
 */
class MainPresenter : MvpPresenter<IMain>() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.i("MainPresenter", "onCreate invoke")
    }

    fun click() {
        view?.changeText()
    }
}